### ISSUE TITLE
pyRevit 4.6.23 not working in Revit 2017 and 2018 - Can not find cpython engines

### ISSUE THREAD

**Issue by salvodragotta** [eirannejad/pyRevit#637]
I'm getting the following error in Revit 2017 and 2018 when installing the latest version (4.6.23).
Revit 2019 & Revit 2020 work as expected. The previous version (4.5 ) was working in Revit 2017 and 2018

**Screenshots**
![image](https://user-images.githubusercontent.com/8536282/60225128-ceaf8000-98c8-11e9-9757-f21c7c3cddd4.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro - 64bit
 - pyRevit Version 4.6.23
 - pyRevit Environment: 
'''
==> Registered Clones (full git repos)
==> Registered Clones (deployed from archive/image)
master | Deploy: "basepublic" | Branch: "master" | Version: "4.6.23" | Path: "C:\Users\rbrown\AppData\Roaming\pyRevit-Master"
==> Attachments
master | Product: "Autodesk Revit 2020 First Customer Ship" | Engine: 277 | Path: "C:\Users\rbrown\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\rbrown\AppData\Roaming\Autodesk\Revit\Addins\2020\pyRevit.addin"
master | Product: "Autodesk Revit 2019.2.1" | Engine: 277 | Path: "C:\Users\rbrown\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\rbrown\AppData\Roaming\Autodesk\Revit\Addins\2019\pyRevit.addin"
Unnamed | Product: "Autodesk Revit 2018.3.3 Security Fix" | Engine: 277 | Path: "C:\ProgramData\pyRevit\pyRevit-v45\pyRevit" | Manifest: "C:\ProgramData\Autodesk\Revit\Addins\2018\pyRevit.addin"
Unnamed | Product: "Autodesk Revit 2017.2.3" | Engine: 277 | Path: "C:\ProgramData\pyRevit\pyRevit-v45\pyRevit" | Manifest: "C:\ProgramData\Autodesk\Revit\Addins\2017\pyRevit.addin"
==> Installed Extensions
==> Default Extension Search Path
C:\Users\rbrown\AppData\Roaming\pyRevit\Extensions
==> Extension Search Paths
==> Extension Sources - Default
https://github.com/eirannejad/pyRevit/raw/master/extensions/extensions.json
==> Extension Sources - Additional
==> Installed Revits
Autodesk Revit 2020 First Customer Ship | Version: 20.0.0.377 | Build: 20190327_2315(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2020"
Autodesk Revit 2019.2.1 | Version: 19.2.10.7 | Build: 20190225_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2019"
Autodesk Revit 2018.3.3 Security Fix | Version: 18.3.3.18 | Build: 20190510_1515(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2018"
Autodesk Revit 2017.2.3 | Version: 17.0.1150.0 | Build: 20171027_0315(x64) | Language: 1033 | Path: "C:\Program Files\Autodesk\Revit 2017"
==> Running Revit Instances
PID: 12212 | Autodesk Revit 2019.2.1 | Version: 19.2.10.7 | Build: 20190225_1515(x64) | Language: 0 | Path: "C:\Program Files\Autodesk\Revit 2019"
==> User Environment
Microsoft Windows 10 [Version 10.0.17134]
Executing User:
Active User: 
Adming Access: Yes
%APPDATA%: "C:\Users\rbrown\AppData\Roaming"
Latest Installed .Net Framework: 4.7.2
No .Net Target Packs are installed.
No .Ne-Core Target Packs are installed.
pyRevit CLI 0.11.0.0
'''

**Comment by eirannejad** [comment]
Open a command prompt and run `pyrevit attach master 277 --installed`. 

`master` is the new pyRevit deplpoyment and is attached to 2019 and 2020
```
master | Product: "Autodesk Revit 2020 First Customer Ship" | Engine: 277 | Path: "C:\Users\rbrown\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\rbrown\AppData\Roaming\Autodesk\Revit\Addins\2020\pyRevit.addin"
master | Product: "Autodesk Revit 2019.2.1" | Engine: 277 | Path: "C:\Users\rbrown\AppData\Roaming\pyRevit-Master" | Manifest: "C:\Users\rbrown\AppData\Roaming\Autodesk\Revit\Addins\2019\pyRevit.addin"
```

But seems like you have an other v45 pyRevit attached to 2017 and 2018
```
Unnamed | Product: "Autodesk Revit 2018.3.3 Security Fix" | Engine: 277 | Path: "C:\ProgramData\pyRevit\pyRevit-v45\pyRevit" | Manifest: "C:\ProgramData\Autodesk\Revit\Addins\2018\pyRevit.addin"
Unnamed | Product: "Autodesk Revit 2017.2.3" | Engine: 277 | Path: "C:\ProgramData\pyRevit\pyRevit-v45\pyRevit" | Manifest: "C:\ProgramData\Autodesk\Revit\Addins\2017\pyRevit.addin"
```

The command above should fix the issue.

**Comment by salvodragotta** [comment]
Thank you for the clue.
Just running the command did not work. 
I had to uninstall v45.

Thanks!

**Comment by Hazen12345** [comment]
Had same problem on some PC's try pyrevit attach master 279 --installed worked on some 277 on others!

**Comment by martinservold** [comment]
I'm got this same error on 4.7.4. now. As stated above it's failing inconsistently, on one computer it will work for 2020 only, and another computer (pathed to the same clone hosted on a server) works for all revits.  
In my particular case I found the issue, I checked to see if it was clashing with any other addins I had by loading only pyrevit, and in the process of doing that I found I had a pyrevit.addin in both my %appdata%\Autodesk\Revit\Addins\20** and %programdata%\Autodesk\Revit\Addins\20**.

Just wanted to post here as something to check in case anyone else runs into this.
